src/celeritas/ext/RootUniquePtr.root.cc

```diff
@@ -29,6 +30,21 @@ void RootWritableDeleter<T>::operator()(T* ptr)
     delete ptr;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Call `TTree->AutoSave()` before deletion. Used by TTree when multiple tfiles
+ * are open at the same time.
+ */
+template<class T>
+void RootAutoSaveDeleter<T>::operator()(T* ptr)
+{
+    CELER_EXPECT(ptr);
+    CELER_LOG(debug) << "Writing " << ptr->ClassName() << " '"
+                     << ptr->GetName() << "'";
+    ptr->AutoSave();
```

Comment:
Does any other interface have `AutoSave`  or just TTree? If the latter you should just make it a template specialization. Or just make separate deleters for ttree and tfile.

Reply:
`AutoSave` is only seen in `TTree`.  `TObject->Write` should probably not be used in general except for `TFile`.   So it might actually indeed make sense to have 2 concrete deletor.  

Side note: Instead `TObject->Write`, especially in a threaded environment one would want to use `dir->WriteObject(ptr)`. i.e. require the writer to specify exactly where to write. (Only `TTree` and Histogram know which file they are attached to/intended for)